tests/baselines/reference/blockScopedBindingUsedBeforeDef.errors.txt

```diff
@@ -1,20 +1,29 @@
 tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(2,12): error TS2448: Block-scoped variable 'a' used before its declaration.
+tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(2,12): error TS2538: Type 'any' cannot be used as an index type.
 tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(5,12): error TS2448: Block-scoped variable 'a' used before its declaration.
+tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(5,12): error TS2538: Type 'any' cannot be used as an index type.
 tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(8,7): error TS2448: Block-scoped variable 'b' used before its declaration.
+tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts(8,7): error TS2538: Type 'any' cannot be used as an index type.
 
 
-==== tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts (3 errors) ====
+==== tests/cases/compiler/blockScopedBindingUsedBeforeDef.ts (6 errors) ====
     // 1:
     for (let {[a]: a} of [{ }]) continue;
                ~
 !!! error TS2448: Block-scoped variable 'a' used before its declaration.
+               ~
+!!! error TS2538: Type 'any' cannot be used as an index type.

```

Comment:
This doesn't feel like a legal error, but it seems like we're only issuing it when there's already an error on the same node. Is there an issue with the error type leaking out into an error message?

Reply:
Most certainly. It's because we choose to consider indexing with `any` an error today (esp. if the thing being indexed has no index signatures), but destructuring accesses don't flag that right now. This PR aligns them.